aya/src/maps/xdp/dev_map_hash.rs

```diff
@@ -0,0 +1,166 @@
+//! An hashmap of network devices.
+
+use std::{
+    borrow::{Borrow, BorrowMut},
+    num::NonZeroU32,
+    os::fd::{AsFd, AsRawFd},
+};
+
+use aya_obj::generated::{bpf_devmap_val, bpf_devmap_val__bindgen_ty_1};
+
+use crate::{
+    maps::{check_kv_size, hash_map, IterableMap, MapData, MapError, MapIter, MapKeys},
+    programs::ProgramFd,
+    sys::{bpf_map_lookup_elem, SyscallError},
+    FEATURES,
+};
+
+use super::dev_map::DevMapValue;
+
+/// An hashmap of network devices.
+///
+/// XDP programs can use this map to redirect to other network
+/// devices.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 5.4.
+///
+/// # Examples
+/// ```no_run
+/// # let mut bpf = aya::Bpf::load(&[])?;
+/// use aya::maps::xdp::DevMapHash;
+///
+/// let mut devmap = DevMapHash::try_from(bpf.map_mut("IFACES").unwrap())?;
+/// let flags = 0;
+/// let ifindex = 32u32;
+/// devmap.insert(ifindex, ifindex, None, flags);
```

Comment:
ifindex ifindex 🤔

Reply:
yeah changed, it does not make any sense.